packages/space-opera/src/components/animation_controls/reducer.ts

```diff
@@ -15,34 +15,22 @@
  *
  */
 
-import {registerStateMutator, State} from '../../space_opera_base.js';
+import {Action, State} from '../../types.js';
 
-/** Set auto play enabled or not */
-const SET_AUTOPLAY_ENABLED = 'SET_AUTOPLAY_ENABLED';
-export const dispatchAutoplayEnabled = registerStateMutator(
-    SET_AUTOPLAY_ENABLED, (state: State, enabled?: boolean) => {
-      state.config = {...state.config, autoplay: !!enabled};
-    });
+const SET_ANIMATION_NAMES = 'SET_ANIMATION_NAMES';
+export function dispatchSetAnimationNames(animationNames: string[]) {
+  return {type: SET_ANIMATION_NAMES, payload: animationNames};
+}
 
-/** Set animation name */
-const SET_ANIMATION_NAME = 'SET_ANIMATION_NAME';
-export const dispatchAnimationName = registerStateMutator(
-    SET_ANIMATION_NAME, (state: State, animationName?: string) => {
-      // Allow animationName === undefined to unset animationName
-      if (animationName && state.animationNames.indexOf(animationName) === -1) {
-        return;
-      }
+export const getAnimationNames = (state: State) =>
```

Comment:
Selectors like this are added and used throughout the code, so we don't have to change every instance of state, whenever we change where a state element is stored.